modules/java/test/android_test/CMakeLists.txt

```diff
@@ -1,77 +1,25 @@
-if(NOT ANT_EXECUTABLE OR NOT ANDROID_EXECUTABLE OR NOT ANDROID_TOOLS_Pkg_Revision GREATER 13)
-  return()
-endif()
+project(opencv_test_android)
 
-project(opencv_test_java)
+set(OPENCV_ANDROID_TEST_DIR "${OpenCV_BINARY_DIR}/android_test" CACHE INTERNAL "")
+file(REMOVE_RECURSE "${OPENCV_ANDROID_TEST_DIR}")
 
-set(OPENCV_JAVA_TEST_DIR "${OpenCV_BINARY_DIR}/android_test" CACHE INTERNAL "")
-file(REMOVE_RECURSE "${OPENCV_JAVA_TEST_DIR}")
-file(REMOVE "${OPENCV_DEPHELPER}/${the_module}_test_source_copy")
+list(APPEND MODULES_TO_TEST "calib3d" "core" "dnn" "features2d" "imgcodecs" "imgproc" "ml" "objdetect" "photo" "video" "videoio")
```

Comment:
There is no need to list modules and iterate over them. All pre-module tests are collected by java bindings generator in build folder in `opencv-build/modules/java_bindings_generator/gen/test/org/opencv/test`. Just add it to to the test project.

Reply:
This folder exists only when building without DANDROID